src/main/java/de/tum/cit/aet/artemis/assessment/service/ResultService.java

```diff
@@ -530,31 +542,91 @@ private Result shouldSaveResult(@NotNull Result result, boolean shouldSave) {
     }
 
     /**
-     * Retrieves aggregated feedback details for a given exercise, calculating relative counts based on the total number of distinct results.
-     * The task numbers are assigned based on the associated test case names, using the set of tasks fetched from the database.
+     * Retrieves paginated and filtered aggregated feedback details for a given exercise.
      * <br>
      * For each feedback detail:
      * 1. The relative count is calculated as a percentage of the total number of distinct results for the exercise.
-     * 2. The task number is determined by matching the test case name with the tasks.
+     * 2. The task numbers are assigned based on the associated test case names. A mapping between test cases and tasks is created using the set of tasks retrieved from the
+     * database.
+     * <br>
+     * Filtering:
+     * - **Search term**: Filters feedback details by the search term (case-insensitive).
+     * - **Test case names**: Filters feedback based on specific test case names (if provided).
+     * - **Task names**: Maps provided task numbers to task names and filters feedback based on the test cases associated with those tasks.
+     * - **Occurrences**: Filters feedback where the number of occurrences (COUNT) is between the provided minimum and maximum values (inclusive).
+     * <br>
+     * Pagination and sorting:
+     * - Sorting is applied based on the specified column and order (ascending or descending).
+     * - The result is paginated based on the provided page number and page size.
      *
-     * @param exerciseId The ID of the exercise for which feedback details should be retrieved.
-     * @return A list of FeedbackDetailDTO objects, each containing:
-     *         - feedback count,
-     *         - relative count (as a percentage of distinct results),
-     *         - detail text,
-     *         - test case name,
-     *         - determined task number (based on the test case name).
+     * @param exerciseId       The ID of the exercise for which feedback details should be retrieved.
+     * @param search           The pageable search DTO containing page number, page size, sorting options, and a search term for filtering results.
+     * @param filterTasks      A list of task numbers to filter the feedback details based on the associated test cases (optional).
+     * @param filterTestCases  A list of test case names to filter the feedback details (optional).
+     * @param filterOccurrence A list of two strings representing the minimum and maximum occurrences to include (optional).
```

Comment:
Why is this a list if you expect exactly two entries? It should either be an array or even two (optional) parameters. I also expect to often only use one of these two, either I'm looking for weird and not yet handled edge cases in the tests or I'm looking for problems that a lot of students have